### ISSUE TITLE
Unable to install nvidia-docker on amazon linux version 1

### ISSUE THREAD

**Issue by kravivar** [NVIDIA/nvidia-docker#738]
Having installation issues on amazon linux, please help.

### 1. Issue or feature description
- I was following the instruction to install nvidia-docker2 using - https://github.com/NVIDIA/nvidia-docker#centos-7-rhel-7475-amazon-linux-12 
- The installation fails with missing docker-ce version

```
Error: Package: nvidia-docker2-2.0.3-1.docker17.09.1.ce.amzn1.noarch (nvidia-docker)
           Requires: docker = 17.09.1ce
           Available: docker-17.12.1ce-1.135.amzn1.x86_64 (amzn-main)
               docker = 17.12.1ce-1.135.amzn1
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

### 2. Steps to reproduce the issue
- Start the amazon linux public AMI - ami-14c5486b or ami-afd15ed0
- yum remove nvidia-docker
- distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
- curl -s -L https://nvidia.github.io/nvidia-docker/$distribution/nvidia-docker.repo | sudo tee /etc/yum.repos.d/nvidia-docker.repo
- yum update
- sudo yum install -y nvidia-docker2

```
LOG OUTPUT:
=============================================================================

[root@ip-10-79-27-77 ~]# yum remove nvidia-docker
Loaded plugins: priorities, update-motd, upgrade-helper
No Match for argument: nvidia-docker
No Packages marked for removal
[root@ip-10-79-27-77 ~]# distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
[root@ip-10-79-27-77 ~]# curl -s -L https://nvidia.github.io/nvidia-docker/$distribution/nvidia-docker.repo | \
>   sudo tee /etc/yum.repos.d/nvidia-docker.repo
[libnvidia-container]
name=libnvidia-container
baseurl=https://nvidia.github.io/libnvidia-container/amzn1/$basearch
repo_gpgcheck=1
gpgcheck=0
enabled=1
gpgkey=https://nvidia.github.io/libnvidia-container/gpgkey
sslverify=1
sslcacert=/etc/pki/tls/certs/ca-bundle.crt

[nvidia-container-runtime]
name=nvidia-container-runtime
baseurl=https://nvidia.github.io/nvidia-container-runtime/amzn1/$basearch
repo_gpgcheck=1
gpgcheck=0
enabled=1
gpgkey=https://nvidia.github.io/nvidia-container-runtime/gpgkey
sslverify=1
sslcacert=/etc/pki/tls/certs/ca-bundle.crt

[nvidia-docker]
name=nvidia-docker
baseurl=https://nvidia.github.io/nvidia-docker/amzn1/$basearch
repo_gpgcheck=1
gpgcheck=0
enabled=1
gpgkey=https://nvidia.github.io/nvidia-docker/gpgkey
sslverify=1
sslcacert=/etc/pki/tls/certs/ca-bundle.crt
[root@ip-10-79-27-77 ~]# sudo yum install -y nvidia-docker2
Loaded plugins: priorities, update-motd, upgrade-helper
libnvidia-container/x86_64/signature                                                                                                             |  455 B  00:00:00     
Retrieving key from https://nvidia.github.io/libnvidia-container/gpgkey
Importing GPG key 0xF796ECB0:
 Userid     : "NVIDIA CORPORATION (Open Source Projects) <cudatools@nvidia.com>"
 Fingerprint: c95b 321b 61e8 8c18 09c4 f759 ddca e044 f796 ecb0
 From       : https://nvidia.github.io/libnvidia-container/gpgkey
libnvidia-container/x86_64/signature                                                                                                             | 2.0 kB  00:00:00 !!! 
nvidia-container-runtime/x86_64/signature                                                                                                        |  455 B  00:00:00     
Retrieving key from https://nvidia.github.io/nvidia-container-runtime/gpgkey
Importing GPG key 0xF796ECB0:
 Userid     : "NVIDIA CORPORATION (Open Source Projects) <cudatools@nvidia.com>"
 Fingerprint: c95b 321b 61e8 8c18 09c4 f759 ddca e044 f796 ecb0
 From       : https://nvidia.github.io/nvidia-container-runtime/gpgkey
nvidia-container-runtime/x86_64/signature                                                                                                        | 2.0 kB  00:00:00 !!! 
nvidia-docker/x86_64/signature                                                                                                                   |  455 B  00:00:00     
Retrieving key from https://nvidia.github.io/nvidia-docker/gpgkey
Importing GPG key 0xF796ECB0:
 Userid     : "NVIDIA CORPORATION (Open Source Projects) <cudatools@nvidia.com>"
 Fingerprint: c95b 321b 61e8 8c18 09c4 f759 ddca e044 f796 ecb0
 From       : https://nvidia.github.io/nvidia-docker/gpgkey
nvidia-docker/x86_64/signature                                                                                                                   | 2.0 kB  00:00:00 !!! 
(1/3): nvidia-container-runtime/x86_64/primary                                                                                                   | 1.4 kB  00:00:00     
(2/3): libnvidia-container/x86_64/primary                                                                                                        | 2.3 kB  00:00:00     
(3/3): nvidia-docker/x86_64/primary                                                                                                              | 1.1 kB  00:00:00     
libnvidia-container                                                                                                                                               10/10
nvidia-container-runtime                                                                                                                                            4/4
nvidia-docker                                                                                                                                                       3/3
Resolving Dependencies
--> Running transaction check
---> Package nvidia-docker2.noarch 0:2.0.3-1.docker17.09.1.ce.amzn1 will be installed
--> Processing Dependency: nvidia-container-runtime = 2.0.0-1.docker17.09.1.amzn1 for package: nvidia-docker2-2.0.3-1.docker17.09.1.ce.amzn1.noarch
--> Processing Dependency: docker = 17.09.1ce for package: nvidia-docker2-2.0.3-1.docker17.09.1.ce.amzn1.noarch
--> Running transaction check
---> Package nvidia-container-runtime.x86_64 0:2.0.0-1.docker17.09.1.amzn1 will be installed
--> Processing Dependency: nvidia-container-runtime-hook < 2.0.0 for package: nvidia-container-runtime-2.0.0-1.docker17.09.1.amzn1.x86_64
--> Processing Dependency: libseccomp for package: nvidia-container-runtime-2.0.0-1.docker17.09.1.amzn1.x86_64
--> Processing Dependency: libseccomp.so.2()(64bit) for package: nvidia-container-runtime-2.0.0-1.docker17.09.1.amzn1.x86_64
---> Package nvidia-docker2.noarch 0:2.0.3-1.docker17.09.1.ce.amzn1 will be installed
--> Processing Dependency: docker = 17.09.1ce for package: nvidia-docker2-2.0.3-1.docker17.09.1.ce.amzn1.noarch
--> Running transaction check
---> Package libseccomp.x86_64 0:2.3.1-2.4.amzn1 will be installed
---> Package nvidia-container-runtime-hook.x86_64 0:1.3.0-1.amzn1 will be installed
--> Processing Dependency: libnvidia-container-tools < 2.0.0 for package: nvidia-container-runtime-hook-1.3.0-1.amzn1.x86_64
--> Processing Dependency: libnvidia-container-tools >= 0.1.0 for package: nvidia-container-runtime-hook-1.3.0-1.amzn1.x86_64
---> Package nvidia-docker2.noarch 0:2.0.3-1.docker17.09.1.ce.amzn1 will be installed
--> Processing Dependency: docker = 17.09.1ce for package: nvidia-docker2-2.0.3-1.docker17.09.1.ce.amzn1.noarch
--> Running transaction check
---> Package libnvidia-container-tools.x86_64 0:1.0.0-0.1.rc.1 will be installed
--> Processing Dependency: libnvidia-container1(x86-64) >= 1.0.0-0.1.rc.1 for package: libnvidia-container-tools-1.0.0-0.1.rc.1.x86_64
--> Processing Dependency: libnvidia-container.so.1(NVC_1.0)(64bit) for package: libnvidia-container-tools-1.0.0-0.1.rc.1.x86_64
--> Processing Dependency: libnvidia-container.so.1()(64bit) for package: libnvidia-container-tools-1.0.0-0.1.rc.1.x86_64
---> Package nvidia-docker2.noarch 0:2.0.3-1.docker17.09.1.ce.amzn1 will be installed
--> Processing Dependency: docker = 17.09.1ce for package: nvidia-docker2-2.0.3-1.docker17.09.1.ce.amzn1.noarch
--> Running transaction check
---> Package libnvidia-container1.x86_64 0:1.0.0-0.1.rc.1 will be installed
---> Package nvidia-docker2.noarch 0:2.0.3-1.docker17.09.1.ce.amzn1 will be installed
--> Processing Dependency: docker = 17.09.1ce for package: nvidia-docker2-2.0.3-1.docker17.09.1.ce.amzn1.noarch
--> Finished Dependency Resolution
Error: Package: nvidia-docker2-2.0.3-1.docker17.09.1.ce.amzn1.noarch (nvidia-docker)
           Requires: docker = 17.09.1ce
           Available: docker-17.12.1ce-1.135.amzn1.x86_64 (amzn-main)
               docker = 17.12.1ce-1.135.amzn1
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
[root@ip-10-79-27-77 ~]# echo $distribution
amzn2018.03
[root@ip-10-79-27-77 ~]#

=============================================================================
```

### 3. Information to [attach](https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/) (optional if deemed irrelevant)

 - [x] Kernel version from `uname -a` - `Linux ip-10-79-27-77 4.14.33-51.37.amzn1.x86_64 #1 SMP Thu May 3 20:07:43 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
 - [x] Driver information from `nvidia-smi -a` - 
```
==============NVSMI LOG==============

Timestamp                           : Thu May 17 22:11:09 2018
Driver Version                      : 390.30

Attached GPUs                       : 1
GPU 00000000:00:1E.0
    Product Name                    : Tesla V100-SXM2-16GB
    Product Brand                   : Tesla
    Display Mode                    : Enabled
    Display Active                  : Disabled
    Persistence Mode                : Enabled
    Accounting Mode                 : Disabled
    Accounting Mode Buffer Size     : 1920
    Driver Model
        Current                     : N/A
        Pending                     : N/A
    Serial Number                   : 0322817061565
    GPU UUID                        : GPU-8e2712fe-6590-250e-8b90-280dbc470b41
    Minor Number                    : 0
    VBIOS Version                   : 88.00.13.00.02
    MultiGPU Board                  : No
    Board ID                        : 0x1e
    GPU Part Number                 : 900-2G503-0000-000
    Inforom Version
        Image Version               : G503.0201.00.03
        OEM Object                  : 1.1
        ECC Object                  : 5.0
        Power Management Object     : N/A
    GPU Operation Mode
        Current                     : N/A
        Pending                     : N/A
    GPU Virtualization Mode
        Virtualization mode         : Pass-Through
    PCI
        Bus                         : 0x00
        Device                      : 0x1E
        Domain                      : 0x0000
        Device Id                   : 0x1DB110DE
        Bus Id                      : 00000000:00:1E.0
        Sub System Id               : 0x121210DE
        GPU Link Info
            PCIe Generation
                Max                 : 3
                Current             : 3
            Link Width
                Max                 : 16x
                Current             : 16x
        Bridge Chip
            Type                    : N/A
            Firmware                : N/A
        Replays since reset         : 0
        Tx Throughput               : 0 KB/s
        Rx Throughput               : 0 KB/s
    Fan Speed                       : N/A
    Performance State               : P0
    Clocks Throttle Reasons
        Idle                        : Active
        Applications Clocks Setting : Not Active
        SW Power Cap                : Not Active
        HW Slowdown                 : Not Active
            HW Thermal Slowdown     : Not Active
            HW Power Brake Slowdown : Not Active
        Sync Boost                  : Not Active
        SW Thermal Slowdown         : Not Active
        Display Clock Setting       : Not Active
    FB Memory Usage
        Total                       : 16160 MiB
        Used                        : 0 MiB
        Free                        : 16160 MiB
    BAR1 Memory Usage
        Total                       : 16384 MiB
        Used                        : 2 MiB
        Free                        : 16382 MiB
    Compute Mode                    : Default
    Utilization
        Gpu                         : 0 %
        Memory                      : 0 %
        Encoder                     : 0 %
        Decoder                     : 0 %
    Encoder Stats
        Active Sessions             : 0
        Average FPS                 : 0
        Average Latency             : 0
    Ecc Mode
        Current                     : Enabled
        Pending                     : Enabled
    ECC Errors
        Volatile
            Single Bit            
                Device Memory       : 0
                Register File       : 0
                L1 Cache            : 0
                L2 Cache            : 0
                Texture Memory      : N/A
                Texture Shared      : N/A
                CBU                 : N/A
                Total               : 0
            Double Bit            
                Device Memory       : 0
                Register File       : 0
                L1 Cache            : 0
                L2 Cache            : 0
                Texture Memory      : N/A
                Texture Shared      : N/A
                CBU                 : 0
                Total               : 0
        Aggregate
            Single Bit            
                Device Memory       : 7
                Register File       : 0
                L1 Cache            : 0
                L2 Cache            : 0
                Texture Memory      : N/A
                Texture Shared      : N/A
                CBU                 : N/A
                Total               : 7
            Double Bit            
                Device Memory       : 0
                Register File       : 0
                L1 Cache            : 0
                L2 Cache            : 0
                Texture Memory      : N/A
                Texture Shared      : N/A
                CBU                 : 0
                Total               : 0
    Retired Pages
        Single Bit ECC              : 1
        Double Bit ECC              : 0
        Pending                     : No
    Temperature
        GPU Current Temp            : 32 C
        GPU Shutdown Temp           : 90 C
        GPU Slowdown Temp           : 87 C
        GPU Max Operating Temp      : 83 C
        Memory Current Temp         : 30 C
        Memory Max Operating Temp   : 85 C
    Power Readings
        Power Management            : Supported
        Power Draw                  : 19.78 W
        Power Limit                 : 300.00 W
        Default Power Limit         : 300.00 W
        Enforced Power Limit        : 300.00 W
        Min Power Limit             : 100.00 W
        Max Power Limit             : 300.00 W
    Clocks
        Graphics                    : 135 MHz
        SM                          : 135 MHz
        Memory                      : 877 MHz
        Video                       : 555 MHz
    Applications Clocks
        Graphics                    : 1312 MHz
        Memory                      : 877 MHz
    Default Applications Clocks
        Graphics                    : 1312 MHz
        Memory                      : 877 MHz
    Max Clocks
        Graphics                    : 1530 MHz
        SM                          : 1530 MHz
        Memory                      : 877 MHz
        Video                       : 1372 MHz
    Max Customer Boost Clocks
        Graphics                    : 1530 MHz
    Clock Policy
        Auto Boost                  : N/A
        Auto Boost Default          : N/A
    Processes                       : None

```

**Comment by flx42** [comment]
Looks like I need to release a new package for Amazon Linux 1. Thanks for the bug report!

**Comment by flx42** [comment]
This is done!

**Comment by abhismatrix3** [comment]
### I am having the same issue, Please help. I am trying to install nvidia-docker2 on sagemaker ml.p2.xlarge instance and followed the steps given for amazon linux. here is the error-

--> Processing Dependency: libnvidia-container.so.1()(64bit) for package: libnvidia-container-tools-1.0.0-0.1.rc.2.x86_64
---> Package nvidia-docker2.noarch 0:2.0.3-1.docker18.03.1.ce.amzn1 will be installed
--> Processing Dependency: docker = 18.03.1ce for package: nvidia-docker2-2.0.3-1.docker18.03.1.ce.amzn1.noarch
--> Running transaction check
---> Package libnvidia-container1.x86_64 0:1.0.0-0.1.rc.2 will be installed
---> Package nvidia-docker2.noarch 0:2.0.3-1.docker18.03.1.ce.amzn1 will be installed
--> Processing Dependency: docker = 18.03.1ce for package: nvidia-docker2-2.0.3-1.docker18.03.1.ce.amzn1.noarch
--> Finished Dependency Resolution
Error: Package: nvidia-docker2-2.0.3-1.docker18.03.1.ce.amzn1.noarch (nvidia-docker)
           Requires: docker = 18.03.1ce
           Installed: docker-17.12.1ce-1.135.amzn1.x86_64 (@amzn-updates)
               docker = 17.12.1ce-1.135.amzn1
           Available: docker-17.03.2ce-1.59.amzn1.x86_64 (amzn-main)
               docker = 17.03.2ce-1.59.amzn1
           Available: docker-17.06.2ce-1.93.amzn1.x86_64 (amzn-updates)
               docker = 17.06.2ce-1.93.amzn1
           Available: docker-17.06.2ce-1.94.amzn1.x86_64 (amzn-updates)
               docker = 17.06.2ce-1.94.amzn1
           Available: docker-17.09.1ce-1.111.amzn1.x86_64 (amzn-updates)
               docker = 17.09.1ce-1.111.amzn1
           Available: docker-17.12.0ce-1.129.amzn1.x86_64 (amzn-updates)
               docker = 17.12.0ce-1.129.amzn1
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

**Comment by aman313** [comment]
@abhismatrix3 Hi. Did you manage to get this to work? Thanks.

**Comment by flx42** [comment]
@aman313 we should have the latest packages up on our repository. If it doesn't work, please file a new issue.

Thanks
